PixiEditor/Models/Tools/Tools/FloodFill.cs

```diff
@@ -23,42 +20,41 @@ public override LayerChange[] Use(Layer layer, Coordinates[] coordinates, Color
             return Only(ForestFire(layer, coordinates[0], color), layer);
         }
 
-        public BitmapPixelChanges ForestFire(Layer layer, Coordinates startingCoords, Color newColor)
+        private BitmapPixelChanges ForestFire(Layer layer, Coordinates startingCoords, Color newColor)
         {
             List<Coordinates> changedCoords = new List<Coordinates>();
 
-            Layer clone = layer.Clone();
-            int width = ViewModelMain.Current.BitmapManager.ActiveDocument.Width;
-            int height = ViewModelMain.Current.BitmapManager.ActiveDocument.Height;
+            int width = layer.Width;
+            int height = layer.Height;
```

Comment:
```suggestion
            int width = ViewModelMain.Current.BitmapManager.ActiveDocument.Width;
            int height = ViewModelMain.Current.BitmapManager.ActiveDocument.Height;
```

Hey, all awesome, but it doesn't work if you don't set the height and width to `ActiveDocument` due to the nature of dynamic bitmaps. They resize to size of content, so for layer without any pixels, it will be 0x0 and nothing will fill :)

Reply:
Oh, didn't know about that - sure, let's change that 👍